de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/allocation/impl/InventoryAllocationDestination.java

```diff
@@ -372,14 +384,55 @@ private I_M_InventoryLine getCreateInventoryLine(final I_M_InOutLine inOutLine,
 		return inventoryLine;
 	}
 
-	private BigDecimal countTUs(final IHUContext huContext, final I_M_HU hu, final I_M_InOutLine receiptLine)
+	private BigDecimal countTUs(final IHUContext huContext, final I_M_HU hu, final I_M_InventoryLine inventoryLine)
 	{
+		// Find out if the hu is VHU or Aggregate -> Move to TU level
+		// Add the TU recursively
+
+		final I_M_HU tuHU;
+		// VHU
+		if (handlingUnitsBL.isVirtual(hu))
+		{
+			tuHU = handlingUnitsDAO.retrieveParent(hu);
+
+		}
+		else
+		{
+			// Aggregated HU
+			tuHU = hu;
+		}
+
+		I_M_InOutLine receiptLine = InterfaceWrapperHelper.create(inventoryLine.getM_InOutLine(), I_M_InOutLine.class);
+
+		final I_M_InOut inout = receiptLine.getM_InOut();
+		final I_C_BPartner partner = inout.getC_BPartner();
+		final I_M_Product product = receiptLine.getM_Product();
+		final Date date = inout.getMovementDate();
+
+		I_M_HU_PI_Item_Product huPIP = hu.getM_HU_PI_Item_Product();
+
+		if (huPIP == null)
+		{
+
+			final I_M_HU_PI_Item materialItem = handlingUnitsDAO
+					.retrievePIItems(tuHU.getM_HU_PI_Version().getM_HU_PI(), partner).stream()
+					.filter(i -> X_M_HU_PI_Item.ITEMTYPE_Material.equals(i.getItemType()))
+					.findFirst().orElse(null);
+
+			if (materialItem != null)
+			{
+				huPIP = huPiItemProductDAO.retrievePIMaterialItemProduct(materialItem, partner, product, date);
+			}
+		}
+
+		inventoryLine.setM_HU_PI_Item_Product(huPIP);
```

Comment:
Also this method's name is `countTUs`, so why do we set the inventoryline's M_HU_PI_Item_Product ? seems to be a rather unexpected sideeffect..